pymc_experimental/model_builder.py

```diff
@@ -225,35 +237,40 @@ def fit(self, data: Dict[str, Union[np.ndarray, pd.DataFrame, pd.Series]] = None
 
         if data is not None:
             self.data = data
-            self._data_setter(data)
-
-        if self.basic_RVs == []:
             self.build()
+            self._data_setter(data)
 
-        with self:
-            self.idata = pm.sample(**self.sample_config)
+        with self.model:
+            if self.sampler_config:
+                self.idata = pm.sample(**self.sampler_config)
```

Comment:
This will work even if `sampler_config` is an empty dict, so you can just remove the if check.